### ISSUE TITLE
Crash after migrating from IOS12 to IOS13

### ISSUE THREAD

**Issue by ovidiuadorian** [project-imas/encrypted-core-data#323]
There seems to be a crash after migrating from IOS 12 to IOS 13. Some things have changed it the app is crashing. 

The error is here:
- (NSDictionary<NSString *,id> *)metadata
{
    NSDictionary<NSString *,id> *superMetadata = super.metadata;

I believe is something related to NSData type that changed his behaviour of description method.

I've investigated but did not found a fix for it till now.

Did someone encountered something the same?

Thx

**Comment by ovidiuadorian** [comment]
I ended up deleting the DB and recreate it at startup

**Comment by jcavar** [comment]
I think the problem is introduction of `NSStoreModelVersionHashesDigest` metadata option in iOS 13. `NSPersistentStore` internally checks if it is set, and tries to load metadata again. This second attempt to load metadata fails because we are already inside of transaction.

**Comment by jcavar** [comment]
I was able to fix it by doing:
`[metadata setValue:@"" forKey:@"NSStoreModelVersionHashesDigest"];`
in `EncryptedStore.m`.

This might not be safe solution, as this is private identifier, so some internal code might rely on it.

**Comment by ovidiuadorian** [comment]
Yeah, they added it and its not documented, I don't know if there are any other implications.
Strange that even looking at "NSStoreModelVersionHashesDigest" on Apple website returns no results.

It seems an internal code
